services/movie_session.py

```diff
@@ -0,0 +1,54 @@
+from django.db.models import QuerySet
+
+from db.models import CinemaHall, Movie, MovieSession
+
+
+def create_movie_session(
+        movie_show_time: str,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+
+    movie = Movie.objects.get(id=movie_id)
+    cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+
+    new_movie_session = MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall=cinema_hall,
+        movie=movie
+    )
+    return new_movie_session
+
+
+def get_movies_sessions(session_date: str = None) -> QuerySet:
+    queryset = MovieSession.objects.all()
+    if session_date:
+        queryset = queryset.filter(show_time__date=session_date)
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: str = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> MovieSession:
+    current_movie = MovieSession.objects.filter(id=session_id)
+    if show_time:
+        current_movie.update(show_time=show_time)
+    if movie_id:
+        movie = Movie.objects.get(id=movie_id)
+        current_movie.update(movie=movie)
+    if cinema_hall_id:
+        cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+        current_movie.update(cinema_hall=cinema_hall)
+
+    return MovieSession.objects.filter(id=session_id)
```

Comment:
The `update_movie_session` function returns a QuerySet instead of a single `MovieSession` instance. Consider using `get` to retrieve the updated instance or ensure the return type matches the function's intended use.